x-pack/plugins/maps/public/classes/layers/ems_vector_tile_layer/ems_vector_tile_layer.tsx

```diff
@@ -242,8 +250,9 @@ export class VectorTileLayer extends TileLayer {
         const newLayerObject = {
           ...layer,
           source:
-            typeof (layer as MbLayer).source === 'string'
-              ? this._generateMbSourceId((layer as MbLayer).source as string)
+            typeof (layer as MbLayer).source === 'string' ||
```

Comment:
This change fixes a regression caused by https://github.com/elastic/kibana/pull/117745. The regression can be viewed by dragging the layer ordering of EMS basemap layer in the web logs sample map. layer.source is undefined for one layer so that case needs to be handled in the typings as well.